src/units/combine.rs

```diff
@@ -215,29 +232,38 @@ mod test {
             }
         ).unwrap();
 
-        // Set all units to stalled, check that the target goes stalled.
-        join!(u1.status(Stalled), u2.status(Stalled), u3.status(Stalled));
-        assert_eq!(t.recv().await, Err(Stalled));
+        // Set one unit to stalled, this triggers picking a new source healthy
+        // with data but there isn’t one, so we go stalled.
+        u1.send_stalled().await;
+        t.recv_stalled().await.unwrap();
+
+        // Now stall the other ones. That shouldn’t change anything.
+        // the last one.
+        u2.send_stalled().await;
+        t.recv_nothing().unwrap();
+        u3.send_stalled().await;
+        t.recv_nothing().unwrap();
 
-        // Set one unit to healthy.
-        u2.data(testrig::update(&[1])).await;
-        u2.status(Healthy).await;
-        assert_eq!(t.recv().await, Err(Healthy));
-        assert_eq!(t.recv().await, Ok(testrig::update(&[1])));
+        // Set one unit to healthy, check that we get an update.
+        u1.send_payload(testrig::update(&[1])).await;
+        assert_eq!(t.recv_payload().await.unwrap(), testrig::update(&[1]));
 
         // Set another unit to healthy. This shouldn’t change anything.
-        u1.data(testrig::update(&[2])).await;
-        u1.status(Healthy).await;
-
-        // Stall them both again.
-        join!(u1.status(Stalled), u2.status(Stalled));
-        assert_eq!(t.recv().await, Err(Stalled));
-
-        // Now unstall one again.
-        u3.data(testrig::update(&[3])).await;
-        u3.status(Healthy).await;
-        assert_eq!(t.recv().await, Err(Healthy));
-        assert_eq!(t.recv().await, Ok(testrig::update(&[3])));
+        u2.send_payload(testrig::update(&[2])).await;
+        t.recv_nothing().unwrap();
+
+        // Now stall the first one and check that we get an update with the
+        // second’s data.
+        u1.send_stalled().await;
+        assert_eq!(t.recv_payload().await.unwrap(), testrig::update(&[2]));
+
+        // Now stall the second one, too, and watch us stall.
```

Comment:
```suggestion
        // Now stall the second one too, and watch us stall.
```

Reply:
Not to get into a grammar discussion, but I have learned way back that there always is a comma before a too.